CMakeLists.txt

```diff
@@ -175,6 +175,24 @@ set(C_PAL_INC_FOLDER
 
 add_subdirectory(build_functions)
 
+if(NOT WIN32)
+    # test for UUID.
+    include (CheckIncludeFile)
+    CHECK_INCLUDE_FILE(uuid.h HAVE_UUID_H)
+    CHECK_INCLUDE_FILE(uuid/uuid.h HAVE_UUID_UUID_H)
+    if (NOT HAVE_UUID_H AND NOT HAVE_UUID_UUID_H)
+        message(FATAL "No UUID implemenation found")
+    elseif(HAVE_UUID_UUID_H)
+        # libuuid implementation
+        add_definitions(-DHAVE_UUID_UUID_H)
+        set(uuid_lib uuid)
+    else()
+        # OSSP uuid
+        add_definitions(-DHAVE_UUID_H)
+        set(uuid_lib ossp-uuid)
+    endif()
+endif()

```

Comment:
Is there a better way to check for ossp-uuid than the existence of uuid.h or uuid/uuid.h? Just thinking that it looks a little bit brittle. Maybe some env var?

Reply:
I did not research by myself, just asking so that maybe we have a second look